extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -534,21 +534,25 @@ export class ProjectsController {
 
 		const result: mssql.SchemaComparePublishProjectResult = await service.schemaComparePublishProjectChanges(operationId, projectPath, fs, utils.getAzdataApi()!.TaskExecutionMode.execute);
 
-		const project = await Project.openProject(projectFilePath);
+		result.success = result.success && result.errorMessage === '';
```

Comment:
This line is the patch for the unfixed underlying issue.  Even if that were fixed, the other changes in this PR would still be bugs.